python/cudf/cudf/core/dataframe.py

```diff
@@ -3040,6 +3040,78 @@ def rename(
         else:
             return out.copy(deep=copy)
 
+    def add_prefix(self, prefix):
+        """
+        Prefix labels with string `prefix`.
+        For DataFrame, the column labels are prefixed.
+        Parameters
+        ----------
+        prefix : str
+            The string to add before each label.
+        Returns
+        -------
+        DataFrame
+            New DataFrame with updated labels.
+        See Also
+        --------
+        DataFrame.add_suffix: Suffix column labels with string `suffix`.
+        Examples
+        --------
+        >>> df = pd.DataFrame({'A': [1, 2, 3, 4], 'B': [3, 4, 5, 6]})
+        >>> df
+           A  B
+        0  1  3
+        1  2  4
+        2  3  5
+        3  4  6
+        >>> df.add_prefix('col_')
+             col_A  col_B
+        0       1       3
+        1       2       4
+        2       3       5
+        3       4       6
+        """
+        out = DataFrame(index=self.index)
+        out._data = self._data.copy(deep=False)
+        out.columns = prefix + self.columns.astype(str)
```

Comment:
`DataFrame.columns` is fairly expensive because it constructs a pandas object. You can use `DataFrame._data.columns` to get a list of column names instead.

Reply:
Thanks a lot for this suggestion!

```python
>>> import cudf
>>> df = cudf.DataFrame({'A': [1, 2, 3, 4], 'B': [3, 4, 5, 6]})
>>> df._data.columns
(<cudf.core.column.numerical.NumericalColumn object at 0x7fa64da782c0>
[
  1,
  2,
  3,
  4
]
dtype: int64, <cudf.core.column.numerical.NumericalColumn object at 0x7fa64da78340>
[
  3,
  4,
  5,
  6
]
dtype: int64)
```
I'm probably missing something, but is there a way to extract column names out of this?

Reply:
ah, try `list(df._data.keys)`

Reply:
```python
>>> df._data.keys
<bound method Mapping.keys of ColumnAccessor(multiindex=False, level_names=(None,))
A: int64
B: int64>
```
will give the column names, but to try to access them:

```python
>>> list(df._data.keys)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'method' object is not iterable
````
Any insights here?

Reply:
Oops, use `keys()` :)

Reply:
Yep this works for me now: `list(df._data.keys())`, thanks @brandon-b-miller!

I guess this will be preferred because constructing a python `list` is cheaper than constructing a `pandas object`?

Reply:
@brandon-b-miller using this, my implementation would change like this:
```python
>>> list(df._data.keys())
['A', 'B']
>>> list(map(lambda os: os + 'test', list(df._data.keys())))
['Atest', 'Btest']
```

OR, there is another option of doing it via list comprehension instead of `map` as done above. 

```python
[s + 'test' for s in list(df._data.keys())]
```

is there any preference from your end on which one would be the best?

Reply:
Listcomp is clearer than map/lambda here IMO.